### PR TITLE
[docs] LCM training

### PR DESCRIPTION
**Pull request by stevhliu**
A first draft of the training docs for LCM. Please feel free to add more details where necessary and suggest any clarifications! ❤️ 

Complementary to the LCM inference docs in #5782

### CODE REVIEW

**Comment by justindujardin** [comment]
Thanks for the quality library, and writing up docs on how to use everything. 

I tried LCM training today (for SD2.1 base) and the script failed to run because the machine didn't have the AWS CLI app installed.

Then after installing aws, it looks like the s3 bucket used in your example isn't publicly accessible. 😓 

Perhaps you want to call out that this is an example of how you *could* run it, but not a runnable example that you can actually use?

One more thing, the scripts are technically broken right now (I guess there are no tests that run for them) because they're using a parse_args argument that isn't defined. I added a guess definition in my local copy to get it to run at all:
```python
    parser.add_argument(
        "--unet_time_cond_proj_dim",
        type=int,
        default=32, # I guessed at this because 32 was used elsewhere in the diffusers codebase
        help="helpful description here",
    )
```

**Comment by patrickvonplaten** [comment]
> Thanks for the quality library, and writing up docs on how to use everything.
> 
> I tried LCM training today (for SD2.1 base) and the script failed to run because the machine didn't have the AWS CLI app installed.
> 
> Then after installing aws, it looks like the s3 bucket used in your example isn't publicly accessible. 😓
> 
> ```
> aws s3 ls s3://muse-datasets/laion-aesthetic6plus-min512-data/
> 
> An error occurred (AccessDenied) when calling the ListObjectsV2 operation: Access Denied
> ```
> 
> Perhaps you want to call out that this is an example of how you _could_ run it, but not a runnable example that you can actually use?
> 
> One more thing, the scripts are technically broken right now (I guess there are no tests that run for them) because they're using a parse_args argument that isn't defined. I added a guess definition in my local copy to get it to run at all:
> 
> ```python
>     parser.add_argument(
>         "--unet_time_cond_proj_dim",
>         type=int,
>         default=32, # I guessed at this because 32 was used elsewhere in the diffusers codebase
>         help="helpful description here",
>     )
> ```

cc @patil-suraj

**Comment by patrickvonplaten** [comment]
> Thanks for the quality library, and writing up docs on how to use everything.
> 
> I tried LCM training today (for SD2.1 base) and the script failed to run because the machine didn't have the AWS CLI app installed.
> 
> Then after installing aws, it looks like the s3 bucket used in your example isn't publicly accessible. 😓
> 
> ```
> aws s3 ls s3://muse-datasets/laion-aesthetic6plus-min512-data/
> 
> An error occurred (AccessDenied) when calling the ListObjectsV2 operation: Access Denied
> ```
> 
> Perhaps you want to call out that this is an example of how you _could_ run it, but not a runnable example that you can actually use?
> 
> One more thing, the scripts are technically broken right now (I guess there are no tests that run for them) because they're using a parse_args argument that isn't defined. I added a guess definition in my local copy to get it to run at all:
> 
> ```python
>     parser.add_argument(
>         "--unet_time_cond_proj_dim",
>         type=int,
>         default=32, # I guessed at this because 32 was used elsewhere in the diffusers codebase
>         help="helpful description here",
>     )
> ```

Hey @justindujardin,

Could you try to open a new chat for this please?

**Comment by justindujardin** [comment]
@patrickvonplaten I opened this issue for it: https://github.com/huggingface/diffusers/issues/5829

You can feel free to move on with this ticket without addressing my comments. 👍

File: docs/source/en/training/lcm_distill.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+write_basic_config()
+```
+
+Lastly, if you want to train a model on your own dataset, take a look at the [Create a dataset for training](create_dataset) guide to learn how to create a dataset that works with the training script.
+
+## Script parameters
+
+<Tip>
+
+The following sections highlight parts of the training script that are important for understanding how to modify it, but it doesn't cover every aspect of the script in detail. If you're interested in learning more, feel free to read through the [script](https://github.com/huggingface/diffusers/blob/main/examples/text_to_image/train_text_to_image_sdxl.py) and let us know if you have any questions or concerns.
```
Comment (pcuenca) [inline docs/source/en/training/lcm_distill.md:68]:
```suggestion
The following sections highlight parts of the training script that are important for understanding how to modify it, but it doesn't cover every aspect of the script in detail. If you're interested in learning more, feel free to read through the [script](https://github.com/huggingface/diffusers/blob/main/examples/consistency_distillation/train_lcm_distill_sd_wds.py) and let us know if you have any questions or concerns.
```

File: docs/source/en/training/lcm_distill.md [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+The training script provides many parameters to help you customize your training run. All of the parameters and their descriptions are found in the [`parse_args()`](https://github.com/huggingface/diffusers/blob/3b37488fa3280aed6a95de044d7a42ffdcb565ef/examples/consistency_distillation/train_lcm_distill_sd_wds.py#L419) function. This function provides default values for each parameter, such as the training batch size and learning rate, but you can also set your own values in the training command if you'd like.
+
+For example, to speedup training with mixed precision using the fp16 format, add the `--mixed_precision` parameter to the training command:
+
+```bash
+accelerate launch train_lcm_distill_sd_wds.py \
+  --mixed_precision="fp16"
+```
+
+Most of the parameters are identical to the parameters in the [Text-to-image](text2image#script-parameters) training guide, so you'll focus on the parameters that are relevant to latent consistency distillation in this guide.
```
Comment (pcuenca) [inline docs/source/en/training/lcm_distill.md:81]:
```suggestion
Most of the parameters are identical to the parameters in the [Text-to-image](text2image#script-parameters) training guide, so we'll focus on the parameters that are relevant to latent consistency distillation in this guide.
```

maybe?

File: docs/source/en/training/lcm_distill.md [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+- `--pretrained_teacher_model`: the path to a pretrained latent diffusion model to use as the teacher model
+- `--pretrained_vae_model_name_or_path`: path to a pretrained VAE; the SDXL VAE is known to suffer from numerical instability, so this parameter allows you to specify a better [VAE](https://huggingface.co/madebyollin/sdxl-vae-fp16-fix) (only applicable if you're distilling a SDXL model)
```
Comment (pcuenca) [inline docs/source/en/training/lcm_distill.md:84]:
```suggestion
- `--pretrained_vae_model_name_or_path`: path to a pretrained VAE; the SDXL VAE is known to suffer from numerical instability, so this parameter allows you to specify an alternative one, like [the one created by madebyollin](https://huggingface.co/madebyollin/sdxl-vae-fp16-fix) to work in `float16` precision.
```

Just a nit, I wouldn't say "better", would mention the author and wouldn't restrict to SDXL.

File: docs/source/en/training/lcm_distill.md [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+- `--w_min` and `--w_max`: the minimum and maximum guidance scale values for guidance scale sampling
+- `--num_ddim_timesteps`: the number of timesteps for DDIM sampling
+- `--loss_type`: the type of loss (L2 or Huber) to calculate for latent consistency distillation
```
Comment (pcuenca) [inline docs/source/en/training/lcm_distill.md:87]: Maybe some guidance here? I think Huber is generally preferred for latent consistency training? /cc @sayakpaul

File: docs/source/en/training/lcm_distill.md [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+    c_top, c_left, _, _ = transforms.RandomCrop.get_params(image, output_size=(resolution, resolution))
+    image = TF.crop(image, c_top, c_left, resolution, resolution)
+    image = TF.to_tensor(image)
+    image = TF.normalize(image, [0.5], [0.5])
+
+    example["image"] = image
+    return example
+```
+
+For improved performance on reading and writing large datasets stored in the cloud, this script uses the [WebDataset](https://github.com/webdataset/webdataset) format to create a preprocessing pipeline to apply transforms and create a dataset and dataloader for training.
```
Comment (pcuenca) [inline docs/source/en/training/lcm_distill.md:108]:
```suggestion
For improved performance on reading and writing large datasets stored in the cloud, this script uses the [WebDataset](https://github.com/webdataset/webdataset) format to create a preprocessing pipeline to apply transforms and create a dataset and dataloader for training. Images are processed and fed to the training loop without having to download the full dataset first.
```

File: docs/source/en/training/lcm_distill.md [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+accelerate launch train_lcm_distill_sd_wds.py \
+    --pretrained_teacher_model=$MODEL_DIR \
+    --output_dir=$OUTPUT_DIR \
+    --mixed_precision=fp16 \
+    --resolution=512 \
+    --learning_rate=1e-6 --loss_type="huber" --ema_decay=0.95 --adam_weight_decay=0.0 \
+    --max_train_steps=1000 \
+    --max_train_samples=4000000 \
+    --dataloader_num_workers=8 \
+    --train_shards_path_or_url='pipe:aws s3 cp s3://muse-datasets/laion-aesthetic6plus-min512-data/{00000..01210}.tar -' \
```
Comment (pcuenca) [inline docs/source/en/training/lcm_distill.md:208]: this is not public, I'd try to use a public wds dataset or clarify.

File: docs/source/en/training/lcm_distill.md [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+pipeline.to("cuda")
+
+prompt = "sushi rolls in the form of panda heads, sushi platter"
+
+image = pipeline(prompt, num_inference_steps=4, guidance_scale=1.0).images[0]
+```
+
+## LoRA
+
+LoRA is a training technique for significantly reducing the number of trainable parameters. As a result, training is faster and it is easier to store the resulting weights because they are a lot smaller (~100MBs). Use the [train_lcm_distill_lora_sd_wds.py](https://github.com/huggingface/diffusers/blob/main/examples/consistency_distillation/train_lcm_distill_lora_sd_wds.py) or [train_lcm_distill_lora_sdxl.wds.py](https://github.com/huggingface/diffusers/blob/main/examples/consistency_distillation/train_lcm_distill_lora_sdxl_wds.py) script to train with LoRA.
```
Comment (pcuenca) [inline docs/source/en/training/lcm_distill.md:240]:
```suggestion
LoRA is a training technique for significantly reducing the number of trainable parameters. As a result, training is faster and it is easier to store the resulting weights because they are a lot smaller (~100MBs). Use the [train_lcm_distill_lora_sd_wds.py](https://github.com/huggingface/diffusers/blob/main/examples/consistency_distillation/train_lcm_distill_lora_sd_wds.py) or [train_lcm_distill_lora_sdxl.wds.py](https://github.com/huggingface/diffusers/blob/main/examples/consistency_distillation/train_lcm_distill_lora_sdxl_wds.py) scripts to train with LoRA.
```

How about linking to the scripts designed for HF datasets? To be included after they are merged?

File: docs/source/en/training/lcm_distill.md [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+
+If you're training on a GPU with limited vRAM, try enabling `gradient_checkpointing`, `gradient_accumulation_steps`, and `mixed_precision` to reduce memory-usage and speedup training. You can reduce your memory-usage even more by enabling memory-efficient attention with [xFormers](../optimization/xformers) and [bitsandbytes'](https://github.com/TimDettmers/bitsandbytes) 8-bit optimizer.
+
+This guide will explore the [train_lcm_distill_sd_wds.py](https://github.com/huggingface/diffusers/blob/main/examples/consistency_distillation/train_lcm_distill_sd_wds.py) script to help you become more familiar with it, and how you can adapt it for your own use-case.
```
Comment (pcuenca) [inline docs/source/en/training/lcm_distill.md:19]: I'd maybe clarify from the onset that this script is designed for `webdataset` datasets, but we can do it when we add the hf datasets version.

**Review by pcuenca — APPROVED** [review]
Awesome, thanks a lot!

**Comment by pcuenca** [comment]
Ref: #5908 for suggested use of public wds datasets.

File: docs/source/en/training/lcm_distill.md (hunk 2 above)
Comment (stevhliu) [reply docs/source/en/training/lcm_distill.md:81]: I prefer "you" because it's the user who is doing the task :)

File: docs/source/en/training/lcm_distill.md (hunk 7 above)
Comment (stevhliu) [reply docs/source/en/training/lcm_distill.md:240]: Yeah lets include after they're merged!

File: docs/source/en/training/lcm_distill.md (hunk 8 above)
Comment (stevhliu) [reply docs/source/en/training/lcm_distill.md:19]: Will do after we've added the other version :)

**Review by patrickvonplaten — APPROVED** [review]
Very cool!
